pkg/aws/cloud.go

```diff
@@ -84,54 +92,67 @@ func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer, logger l
 
 		if region == "" {
 			err := (error)(nil)
-			region, err = metadata.Region()
+			region, err = ec2Metadata.Region()
 			if err != nil {
 				return nil, errors.Wrap(err, "failed to introspect region from EC2Metadata, specify --aws-region instead if EC2Metadata is unavailable")
 			}
 		}
 		cfg.Region = region
 	}
-	awsCFG := aws.NewConfig().WithRegion(cfg.Region).WithSTSRegionalEndpoint(endpoints.RegionalSTSEndpoint).WithMaxRetries(cfg.MaxRetries).WithEndpointResolver(endpointsResolver)
-	opts = session.Options{}
-	opts.Config.MergeIn(awsCFG)
-	if !hasIPv4 {
-		opts.EC2IMDSEndpointMode = endpoints.EC2IMDSEndpointModeStateIPv6
-	}
-	sess := session.Must(session.NewSessionWithOptions(opts))
-	injectUserAgent(&sess.Handlers)
+	awsConfig, err := config.LoadDefaultConfig(context.TODO(),
```

Comment:
1. cfg.AWSEndpoints seems dropped support? It's a flag used to override api endpoints (e.g. FIPS ones).
2. seems endpoints.RegionalSTSEndpoint is not explicitly specified, is it the default behavior of AWSSDKv2? We don't want to use global STS endpoint.

Reply:
I have verified that the GO SDK v2 uses regional STS endpoints by default. https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html#:~:text=SDK%20for%20Go%20V2,Request%20failure

Good catch. I have added support for custom AWSEndpoints now in recent commit. Please take a look.